### PR TITLE
[INLONG-5362][Agent] Supports the collection of data splicing metadata information

### LINKED ISSUE #5362: [Feature][Agent] Supports the collection of data splicing metadata information
### Description

In some scenarios, data needs to be spliced together with metadata information.
For example:
image

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by GanfengTan**
The feature supports the collection of data splicing metadata information.

- Fixes #5362 

### Motivation

1.Refactoring FileReader
2.Add AbstractReader
3.Connect k8s metadata

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [x] This change added tests and can be verified as follows:

*TestTextFileReader

### CODE REVIEW

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/KubernetesConstants.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package org.apache.inlong.agent.constant;
+
+/**
+ * k8s information
+ */
+public class KubernetesConstants {
+
+    public static final String HTTPS = "https://";
```
Comment (healchow) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/KubernetesConstants.java:26]: Only support `https`?

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import static org.apache.inlong.agent.constant.KubernetesConstants.NAMESPACE;
+import static org.apache.inlong.agent.constant.KubernetesConstants.POD_NAME;
+
+/**
+ * k8s file reader
+ */
+public final class KubernetesFileReader extends AbstractFileReader {
+
+    private static final Logger log = LoggerFactory.getLogger(KubernetesFileReader.class);
+    public FilesReader filesReader;
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java:53]: This field was already defined in its parent class, no need to declare it again here.

Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java]: Yes, you are right.

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/KubernetesConstants.java (hunk 1 above)
Comment (GanfengTan) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/KubernetesConstants.java:25]: k8s config:
image

**Review by EMsnap — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Review by pocozh — APPROVED** [review]
+1

**Pull request merged**